### PR TITLE
Issue #51: Added support for feature policy header

### LINKED ISSUE #51: Add support for feature policy header
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy

### PR DESCRIPTION
**Pull request by Peterburnett**
Closes #51

### CODE REVIEW

File: classes/helper.php [hunk 1]
```diff
@@ -41,6 +41,11 @@ class helper {
      */
     private static $bootstrapped = false;
 
+    /**
+     * @var boolean Have we sent the feature policy headers already?
+     */
+    private static $featurebootstrapped = false;
```
Comment (brendanheywood) [inline classes/helper.php]: do you need this? headers are queued and by default they replace previously set headers until they get flushed

```
header ( string $header [, bool $replace = TRUE
```

File: classes/helper.php [hunk 2]
```diff
@@ -61,5 +66,17 @@ public static function enable_csp_header() {
...
-}
 
+    public static function enable_feature_policy() {
+        $settings = get_config('local_csp');
+        if (self::$featurebootstrapped or empty($settings->feature_policy_enable)) {
+            return;
+        }
+        self::$featurebootstrapped = true;
+
+        $featureheader = trim(str_replace(array("\r\n", "\r"), " ", $settings->feature_policy));
```
Comment (brendanheywood) [inline classes/helper.php:71]: If it is one entry per line I think we need to lock down the delimiter

ie  a policy like this:

```
Feature-Policy: microphone 'none'; geolocation 'none'
```

would have a config value of
```
microphone 'none'
geolocation 'none'
```

so we should add in the `;`

either way should be explicitly in the setting help text

Comment (Peterburnett) [reply classes/helper.php:71]: Hmm perhaps I worded that badly. The semicolon is always required, but it makes no difference whether they are chained on one line, or seperated by a line break

File: classes/helper.php (hunk 1 above)
Comment (Peterburnett) [reply classes/helper.php]: Yeah its not exactly necessary. Its slightly more efficient and matches the CSP header behaviour of only being sent once no matter what, but I can remove it. These headers should remain the same in pretty much every case.

File: lang/en/local_csp.php [hunk 3]
```diff
@@ -41,6 +41,10 @@
 $string['cspheaderreporting'] = 'Content-Security-Policy-Report-Only';
 $string['cspheaderreportinghelp'] = 'Monitor and report CSP violations';
 $string['csphttpresponseheader'] = 'CSP HTTP response header';
+$string['enablefeaturepolicy'] = 'Enable Feature-Policy header';
+$string['enablefeaturepolicydescription'] = 'Send a feature policy header as part of the Plugin headers sent. This header controls what browser features are allowed to be accessed by DOM elements.';
+$string['featurepolicy'] = 'Feature-Policy';
```
Comment (brendanheywood) [inline lang/en/local_csp.php]: should this be translated?

**Pull request merged**